src/App.jsx

```diff
@@ -14,33 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABETICALLY = 'alphabetically';
+const SORT_BY_LENGTH = 'by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  let visibleGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    visibleGoods = visibleGoods.toReversed();
+  }
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_ALPHABETICALLY:
+        visibleGoods = visibleGoods.toSorted();
+        if (reversed) {
+          visibleGoods = visibleGoods.toReversed();
+        }
+
+        break;
+      case SORT_BY_LENGTH:
+        if (reversed) {
+          visibleGoods = visibleGoods.toSorted((a, b) => b.length - a.length);
+        } else {
+          visibleGoods = visibleGoods.toSorted((a, b) => a.length - b.length);
+        }
+
+        break;
+      default:
+        return null;
+    }
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SORT_ALPHABETICALLY,
+          })}
+          onClick={() => setSortField(SORT_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
+          Reverse
+        </button>
+
+        {sortField || reversed
```

Comment:
usually, conditions like that are moved to a separate variable. Because if we need to add another condition - it gets more difficult to look for it inside the return statement.

I'd say something like `const canReset = sortField || reversed;` and then use it here.

But again, treat it as a minor suggestion, no big deal.